### ISSUE TITLE
Error logs for failed experiment run

### ISSUE THREAD

**Issue by sourabhpoddar404** [dice-group/gerbil#393]
I have configured experiment on Gerbil for QA system. Experiment is running for some time (15 min) then failing with error "The annotator caused too many single errors". I have not been able to find any error log on our QA system. 
Is it possible to find the error log from Gerbil?
Experiment Id : 202107200001

I have followed the response structure suggested for running experiment on Gerbil. Please suggest if it is possible to find any logs of the experiment from Gerbil end?

**Comment by MichaelRoeder** [comment]
The format of the answer does not seem to be correct:
```
2021-07-20 12:55:13,259 [pool-30-thread-20] ERROR [org.aksw.gerbil.annotator.impl.qa.ExtendedQALDBasedWebService] - <Couldn't parse the response.>
com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "answers" (class org.aksw.qa.commons.load.json.QaldJson), not marked as ignorable (2 known properties: "dataset", "questions"])
 at [Source: (ByteArrayInputStream); line: 1, column: 13] (through reference chain: org.aksw.qa.commons.load.json.QaldJson["answers"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:61) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnknownProperty(DeserializationContext.java:823) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1153) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1589) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1567) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:294) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4014) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3071) ~[jackson-databind-2.9.10.5.jar:2.9.10.5]
	at org.aksw.qa.commons.load.json.ExtendedQALDJSONLoader.readJson(ExtendedQALDJSONLoader.java:115) ~[commons-0.4.22.jar:0.4.22]
	at org.aksw.gerbil.annotator.impl.qa.ExtendedQALDBasedWebService.answerQuestion(ExtendedQALDBasedWebService.java:110) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.TimeMeasuringAnnotatorDecorator.performQATask(TimeMeasuringAnnotatorDecorator.java:276) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.TimeMeasuringAnnotatorDecorator$TimeMeasuringQASystem.answerQuestion(TimeMeasuringAnnotatorDecorator.java:205) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.ErrorCountingAnnotatorDecorator.performQATask(ErrorCountingAnnotatorDecorator.java:401) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.ErrorCountingAnnotatorDecorator$ErrorCountingQASystem.answerQuestion(ErrorCountingAnnotatorDecorator.java:213) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.SingleInstanceSecuringAnnotatorDecorator.performQATask(SingleInstanceSecuringAnnotatorDecorator.java:352) [classes/:?]
	at org.aksw.gerbil.annotator.decorator.SingleInstanceSecuringAnnotatorDecorator$SingleInstanceSecuringQASystem.answerQuestion(SingleInstanceSecuringAnnotatorDecorator.java:211) [classes/:?]
	at org.aksw.gerbil.execute.ExperimentTask.runExperiment(ExperimentTask.java:530) [classes/:?]
	at org.aksw.gerbil.execute.ExperimentTask.run(ExperimentTask.java:167) [classes/:?]
	at org.aksw.simba.topicmodeling.concurrent.workers.WorkerImpl.run(WorkerImpl.java:44) [topicmodeling.concurrent-0.0.2-20160315.131022-1.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_191]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_191]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_191]
2021-07-20 12:55:13,260 [pool-30-thread-20] ERROR [org.aksw.gerbil.annotator.decorator.ErrorCountingAnnotatorDecorator] - <Got an Exception from the annotator (Porque_libre_Tebaqa (WS)): Couldn't parse the response.>
```

**Comment by sourabhpoddar404** [comment]
Thanks it was helpful I was able to run the experiment. I would like to point out that I structured the response based on template QAsystem provided on Gerbil git page "https://github.com/dice-group/GerbilQA-Benchmarking-Template". But evidently it doesn't work.

**Comment by RicardoUsbeck** [comment]
Hi, could you describe the error or what you have had to change?

**Comment by sourabhpoddar404** [comment]
Gerbil expects the response from QA to be in QALD format, but I structured it earlier according to this template provided in Gerbil git page "https://github.com/dice-group/GerbilQA-Benchmarking-Template". Experiment was running for approx 15 min then getting failed with error ""The annotator caused too many single errors". After I got the error log commented by Michael I changed the response of my QA system according to QALD format and it worked.
